qiita_db/commands.py

```diff
@@ -300,3 +305,123 @@ def load_parameters_from_cmd(name, fp, table):
                          "The format is PARAMETER_NAME<tab>VALUE")
 
     return constructor.create(name, **params)
+
+
+def update_preprocessed_data_from_cmd(sl_out_dir, study_id, ppd_id=None):
+    """Updates the preprocessed data of the study 'study_id'
+
+    Parameters
+    ----------
+    sl_out_dir : str
+        The path to the split libraries output directory
+    study_id : int
+        The study_id of the study to be updated
+    ppd_id : int
+        The id of the preprocessed_data to be updated. If not provided, the
+        preprocessed data with the lowest id in the study will be updated.
+
+    Returns
+    -------
+    qiita_db.PreprocessedData
+        The updated preprocessed data
+
+    Raises
+    ------
+    IOError
+        If sl_out_dir does not contain all the required files
+    ValueError
+        If the study does not have any preprocessed data
+        If ppd_id is provided and it does not belong to the given study
+
+    Notes
+    -----
+    If the study has more than one preprocessed data, the one with the lowest
+    id is updated.
+    """
+    # Check that we have all the required files
+    path_builder = partial(join, sl_out_dir)
+    new_fps = {'preprocessed_fasta': path_builder('seqs.fna'),
+               'preprocessed_fastq': path_builder('seqs.fastq'),
+               'preprocessed_demux': path_builder('seqs.demux'),
+               'log': path_builder('split_library_log.txt')}
+
+    missing_files = [key for key, val in viewitems(new_fps) if not exists(val)]
+    if missing_files:
+        raise IOError(
+            "The directory %s does not contain the following required files: "
+            "%s" % (sl_out_dir, ', '.join(missing_files)))
+
+    # Get the preprocessed data to be updated
+    study = Study(study_id)
+    ppds = study.preprocessed_data()
+    if not ppds:
+        raise ValueError("Study %s does not have any preprocessed data")
+
+    if ppd_id:
+        if ppd_id not in ppds:
+            raise ValueError("The preprocessed data %d does not exist in "
+                             "study %d." % (ppd_id, study_id))
```

Comment:
Unless we foresee there being way too many preprocessed data identifiers, it might be useful to print the available ids as part of this error. I understand if we don't, but it would be nice *i think*.